webwork/webwork_to_md.py

```diff
@@ -0,0 +1,120 @@
+import yaml
+import re
+
+# open file
+problem_file_name = 'NU_U17-2-07-010.pg'  # short answer
+path = '../../webwork-open-problem-library/Contrib/BrockPhysics/College_Physics_Urone/2.Kinematics/' + problem_file_name
+file = open(path, 'r')
+file_contents = file.read()
+
+# declare variables
+metadata = "## "
+chapter_src = "DBchapter"
+section_src = "DBsection"
+author_src = "AuthorText1"
+editor_src = "Editor"
+keywords_src = "KEYWORDS"
+date_src = "Date"
+
+title = author = editor = date = source = problem_type = outcomes = assets = server = ""
+tags = []
+
+# get metadata from file
+for item in file_contents.split("\n"):
+    if metadata + chapter_src in item:
+        title = item[item.find("(") + 1:item.find(")")].replace("'", "")
+    if metadata + section_src in item:
+        title += " - " + item[item.find("(") + 1:item.find(")")].replace("'", "")
+    if metadata + author_src in item:
+        author = item[item.find("(") + 1:item.find(")")].replace("'", "")
+    if metadata + editor_src in item:
+        editor = item[item.find("(") + 1:item.find(")")].replace("'", "")
+    if keywords_src in item:
+        tags = item[item.find("(") + 1:item.find(")")].replace("'", "").replace(",", "").split()
+    if metadata + date_src in item:
+        date = item[item.find("(") + 1:item.find(")")].replace("'", "")
+
+# get problem text from file
+start_of_problem_src = "BEGIN_TEXT"
+end_of_problem_src = "END_TEXT"
+begin_hint_src = "BEGIN_HINT"
+
+problem_full = file_contents[file_contents.index(start_of_problem_src):file_contents.index(begin_hint_src)]
+problem_header = problem_full \
+    .replace(' \\', '') \
+    .replace('\\', '') \
+    .replace('/', '') \
+    .replace('<strong>', '') \
+    .replace('$', '') \
+    .replace('{', '') \
+    .replace('}', '') \
+    .replace('PAR', '') \
+    .replace('textrm', '') \
+    .strip()
+# find all problem text between
+problem_multi_para = problem_header.replace(end_of_problem_src, "").replace(start_of_problem_src, "")
+
+# remove ans_rule line from problem
+problem_clean_up = re.sub(r"ans_rule(.+?[(])(.+?[)])", '', problem_multi_para)
+# remove ANS line from problem
+problem_no_ans = re.sub(r"ANS(\(.+?[?<!)]\));", '', problem_clean_up)
+# remove empty lines from problem
+problem_no_empty_lines = os.linesep.join([s for s in problem_no_ans.splitlines() if s])
+# remove hint lin from beginning of problem
+problem_text = problem_no_empty_lines[problem_no_empty_lines.index("hint.")+6:]
+
+# get answer from file
+start_of_answer_src = "showHint"
+end_of_answer_src = "BEGIN_TEXT"
+
+answer_with_hint = file_contents[file_contents.index(start_of_answer_src):file_contents.index(end_of_answer_src)]
+answer_section = re.sub(rf"{start_of_answer_src} =\d+;", '', answer_with_hint)\
+    .replace(end_of_answer_src, '')\
+    .replace(';', '')\
+    .replace('$', '').strip()
+if "random" in answer_section:
+    answer_section = "from random import randrange\n" + answer_section.replace('random', 'randrange')
```

Comment:
Here I suggest using a regex matching "variable = random(...)".  Current this is too greedy.  For example, file I'm looking at has:
```
$x0 = random(-10,10);
$y0 = non_zero_random(-10,10);
$z0 = non_zero_random(-10,10);
```

A more conservative approach is explicitly match lines.  And lines that are not explicitly and conservatively translated get comments like ` # TODO port me` appended.

Reply:
Thanks for the suggestions! The answer section is still WIP but I will definitely keep this in mind :)